cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -2840,37 +2798,68 @@ pub(crate) mod tests {
     }
 
     #[tokio::test]
-    async fn locking_message_is_passed_through() {
-        let manifest = &*TEST_TYPED_MANIFEST;
+    async fn locking_general_message() {
+        let manifest = Manifest::from_str(indoc! {r#"
+                version = 1
 
-        let mut client = catalog::MockClient::new();
-
-        for response in [
-            TEST_RESOLUTION_RESPONSE_UNKNOWN_MSG.clone(),
-            TEST_RESOLUTION_RESPONSE_GENERAL.clone(),
-            TEST_RESOLUTION_RESPONSE_SYSTEMS_NOT_ON_SAME_PAGE.clone(),
-        ] {
-            // It doesn't make sense to generate mocks since:
-            // - We're testing for future messages with unknown message
-            // - General isn't currently used serverside
-            // We could probably generate one for SYSTEMS_NOT_ON_SAME_PAGE if we
-            // wanted.
-            let response_msg: ResolutionMessage =
-                response.first().unwrap().msgs.first().unwrap().clone();
-            client.push_resolve_response(response);
-
-            let locked_manifest =
-                Lockfile::resolve_manifest(manifest, None, &client, &InstallableLockerMock::new())
-                    .await;
-            if let Err(ResolveError::ResolutionFailed(res_failures)) = locked_manifest {
-                // A newline is added for formatting when it's a single message
-                assert_eq!(
-                    res_failures.to_string(),
-                    format!("\n{}", response_msg.msg())
-                );
+                [install]
+                ps.pkg-path = "darwin.ps"
+
+                [options]
+                systems = [ "x86_64-darwin", "aarch64-linux" ]
+            "#})
+        .unwrap();
+
+        let client = catalog_replay_client(
+            GENERATED_DATA.join("resolve/darwin_ps_incompatible_transform_error_to_general.yaml"),
+        )
+        .await;
```

Comment:
the docs on `catalog_replay_client` state: 

> Tests must be run with #[tokio::test(flavor = "multi_thread")]

looks like that is not necessarily true?